project.nix

```diff
@@ -34,7 +34,17 @@ let
   });
 
   sources = {
-    neuron = gitignoreSource ./.;
+    neuron = nix-filter {
+      root = ./.;
+      name = "neuron";
+      include = [
+        "neuron-search"
+        "neuron.cabal"
+        (inDirectory "exe")
+        (inDirectory "src")
+        (inDirectory "test")
+      ];
```

Comment:
While by including only these paths, both `nix build` and `nix-build` finishes without any error, I am not really familiar with the internals of the project. There might be other files required for some reason, so it is best if you double check this part.